### DISCUSSION TITLE
Raw SQL for Custom Functions

### DISCUSSION THREAD

**Discussion by xyzkaa** [collerek/ormar#242] [Q&A]
Is there a way we can execute raw SQL using ormar? We added an extension to our PostgreSQL database that utilizes custom functions, which are not supported by out of the box ormar functions. 

Afterwards, I would like to perform other ormar functions on these results, such as `.filter()`, etc.

Thank you!

**Comment by collerek** [comment]
Yes I plan to add this functionality in the future, right now it's not possible directly through ormar but you can always take the database (same you use in Meta) and call `execute()`.

Even if there gonna be a custom sql support, unless it's gonna be semi-custom (like f-functions, some kind of variation of the model fields (like count from column etc.), something similar to one described in #227 ) there probably wont be a support for `filter()` etc. on the result of that call. Using ormar functions require presence of ormar fields, and when you execute the custom sql you can return anything you want from any table you want. Moreover if you are already executing a custom sql you can simply add a where clause there.
